langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -2588,6 +2588,96 @@ function testCharClassesWithMultipleRangesAndAtoms() returns error? {
     assertEquality("b", (<regexp:Span>res9).substring());
 }
 
+function testRegexpWithUnicodeChars() {
+    string startChar = string `\u{0d9c}`;
+    string endChar = string `\u{1F310}`;
+    regexp:Span? resA1 = regexp:find(re = re `\u{1F30D}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA1 is regexp:Span);
+    regexp:Span? resA2 = regexp:find(re `\u{1F30D}`, "🌍😀 Hello! ❤️", 5);
+    assertTrue(resA2 is ());
+    regexp:Span? resA3 = regexp:find(re `\u{1F30D}`, "😀 Hello 🌍! ❤️", 6);
+    assertTrue(resA3 is regexp:Span);
+    regexp:Span? resA4 = regexp:find(re `\u{1F31F}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resA4 is ());
+    regexp:Span? resA5 = regexp:find(re `\u{0d9c}`, "පරිගණක 10");
+    assertTrue(resA5 is regexp:Span);
+    regexp:Span? resA6 = regexp:find(re `[${startChar}${endChar}]`, "😀 Hello, 🌏! ❤️");
+    assertTrue(resA6 is regexp:Span);
+    regexp:Span? resA7 = regexp:find(re `[${startChar}${endChar}]`, "😀 Hello,! ❤️");
+    assertTrue(resA7 is regexp:Span);
+    // Invalid code point range
+    regexp:Span? resA8 = regexp:find(re `\u{FF3AD}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resA8 is ());
+
+    regexp:Span[] resB1 = regexp:findAll(re `\u{1F30D}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resB1.length() == 1);
+    regexp:Span[] resB2 = regexp:findAll(re `\u{1F30D}`, "🌍😀 Hello! ❤️", 5);
+    assertTrue(resB2.length() == 0);
+    regexp:Span[] resB3 = regexp:findAll(re `\u{1F30D}`, "😀 Hello 🌍! 🌍❤️", 6);
```

Comment:
Shall we do a assertion for start index, end index for tis tst.
This is because since we have issues related to code points and indexes, when the strings have emojies

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/40368/commits/d10a307642318c20ec015c816bc8541bb2e5a572.